lib/pure/httpclient.nim

```diff
@@ -840,13 +841,21 @@ proc parseResponse(client: HttpClient | AsyncHttpClient,
     else:
       # Parse headers
       var name = ""
-      var le = parseUntil(line, name, ':', linei)
-      if le <= 0: httpError("invalid headers")
-      inc(linei, le)
-      if line[linei] != ':': httpError("invalid headers")
-      inc(linei) # Skip :
+      var leadingSpaces = skipWhitespace(line, linei)
+      if leadingSpaces == 0:
+        var le = parseUntil(line, name, ':', linei)
+        if le <= 0 or le >= line.len: httpError("invalid headers")
```

Comment:
`le >= line.len` makes `httpclient` a little more stricter regarding random lines it finds in the response - basically, if the response is like:
```
Header-One: value
Header-Two: value
somerandomname
```

`httpclient` will error out on `somerandomname`. Chromium and Firefox don't error out on that and just skip the line - should we follow them or throw an error?